proposals/010-authorizer.md

```diff
@@ -0,0 +1,212 @@
+
+The table below sets out the authorization checks and filters will be implemented.
+
+| Operation | Resource Type | Kafka Message                                                                                                                                                                                                          |
+|-----------|---------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| READ      | Topic         | Fetch, ShareFetch, ShareGroupFetch, ShareAcknowledge, AlterShareGroupOffsets, DeleteShareGroupOffsets, OffsetCommit, TxnOffsetCommit, OffsetDelete                                                                     |
+| WRITE     | Topic         | Produce, InitProducerId, AddPartitionsToTxn                                                                                                                                                                            |
+| CREATE    | Topic         | CreateTopics                                                                                                                                                                                                           |
+| DELETE    | Topic         | DeleteTopics                                                                                                                                                                                                           |
+| ALTER     | Topic         | AlterConfigs, IncrementalAlterConfigs, CreatePartitions                                                                                                                                                                |
+| DESCRIBE  | Topic         | ListOffset, OffsetFetch, OffsetFetchForLeaderEpoch DescribeProducers, ConsumerGroupHeartbeat, ConsumerGroupDescribe, ShareGroupHeartbeat, ShareGroupDescribe, MetaData, DescribeTopicPartitions, ConsumerGroupDescribe |
+| CONNECT   | Cluster       | SaslAuthenticate                                                                                                                                                                                                       |
+
+In general, the filter will make access decisions in the same manner as Kafka itself.  This means it will apply the same authorizer checks that Kafka enforces itself and generate error responses in the same way.  It will
+From the client's perspective, it will be impossible for it to distinguish between the proxy and kafka cluster itself.
+It will also use the same implied operation semantics as implemented by Kafka itself, such as where `ALTER` implies `DESCRIBE`, as described by 
+`org.apache.kafka.common.acl.AclOperation`.
```

Comment:
We should define what has responsibility for this. It could be:

* A part of the contract of `interface Authorizer`
* An implementation detail of the particular Authorizer implementation being proposed, so that other Authorizers are not required to respect this implication

If the former then it's natural to model the implication relation on the operation enum. If the latter then probably we're saying that the Authorizer implementation has to special case those topic operations.